database/redis/database_test.go

```diff
@@ -28,3 +28,36 @@ func TestInitialization(t *testing.T) {
 		So(err, ShouldNotBeNil)
 	})
 }
+
+func TestAllowStale(t *testing.T) {
+	logger, _ := logging.ConfigureLog("stdout", "info", "test")
+
+	Convey("Allow stale", t, func() {
+		Convey("When using redis directly, returns same db", func() {
+			database := newTestDatabase(logger, emptyConfig)
+
+			staleDatabase := database.AllowStale()
+
+			So(staleDatabase, ShouldPointTo, database)
+		})
+
+		Convey("When using sentinel, returns slave db instance, retains references", func() {
```

Comment:
Consider adding a test case when Sentinel is configured, but there are no slaves, only master.

Reply:
I'm also curious, what happens when there is a live slave at application startup, but then it goes offline. Does this code fallback to using master without restart?

What about two slaves? If one of them goes offline, does this code reconnect to the other?

Reply:
Dropping a faulted connection and establishing a new one seems to be redigo's responsibility. The `Dial()` in database_dialer.go queries sentinel and returns a valid connection. My guess is that when a connection breaks, redigo will drop it from pool and call `Dial()` to obtain a new one.

Reply:
Also, I've added a testcase for when the slave reads are disabled